add --set-perms to pysogs, let --add-room also use them

tewinget opened the pull request.

tewinget committed:
add --set-perms to pysogs, let --add-room also use them

jagerman commented on sogs/__main__.py:
```diff
@@ -51,6 +57,27 @@
+)
+ap.add_argument(
+    "--add-perms",
+    help="With --add-room or --set-perms, set these permissions to true; --add-perms=[r][w][u][a]",
```
```suggestion
    help="With --add-room or --set-perms, add these permissions to; should be a string of 1-4 of the letters \"rwua\" for [r]ead, [w]rite, [u]pload, and [a]ccess.",
```

jagerman commented on sogs/__main__.py:
```diff
@@ -51,6 +57,27 @@
+actions.add_argument('--set-perms', help="Sets permissions for the room given by --rooms")
+ap.add_argument(
+    '--users',
+    help="Users to set permissions for with --set-perms; if omitted, set global perms",
```
```suggestion
    help="Users to set permissions for with --set-perms; if omitted, set the default permissions for the room.",
```

jagerman commented on sogs/__main__.py:
```diff
@@ -51,6 +57,27 @@
+)
+ap.add_argument(
+    "--remove-perms",
+    help="With --add-room or --set-perms, set these permissions to false; "
+    "--remove-perms=[r][w][u][a]",
```
```suggestion
    help="With --add-room or --set-perms, deny these permissions; should be a string of 1-4 of the letters \"rwua\" for [r]ead, [w]rite, [u]load, and [a]ccess.",
```

jagerman commented on sogs/__main__.py:
```diff
@@ -51,6 +57,27 @@
+)
+ap.add_argument(
+    "--clear-perms",
+    help="With --add-room or --set-perms, clear room or user overrides on these permissions; "
+    "--clear-perms=[r][w][u][a]",
```
```suggestion
    help="With --add-room or --set-perms, clear room or user overrides on these permissions; should be a string of 1-4 of the letters \"rwua\".",
```

jagerman commented on sogs/__main__.py:
```diff
@@ -51,6 +57,27 @@
+    metavar='SESSIONID',
+)
+ap.add_argument(
+    "--add-perms",
```
Thoughts on calling this "--grant-perms" instead of "--add-perms"?  (And likewise revoke instead of remove)?

tewinget replied:
utter indifference

jagerman committed:
Option wording tweaks

jagerman committed:
Print default room permissions in room summary

jagerman committed:
Fix u/u2 code

Don't need to worry about getting a blinded user here; the User
constructor does the right thing already.

Also fixes a bug where `users` was a list of lists instead of a list.
